storage/src/main/java/org/apache/kafka/storage/internals/log/LogValidator.java

```diff
@@ -574,7 +574,7 @@ private static Optional<ApiRecordError> validateRecord(RecordBatch batch,
         if (batch.magic() <= RecordBatch.MAGIC_VALUE_V1 && batch.isCompressed()) {
             try {
                 record.ensureValid();
-            } catch (InvalidRecordException e) {
```

Comment:
In order to keep consistency, we should rethrow  `CorruptRecordException` here. for example:
```java
            } catch (CorruptRecordException e) {
                metricsRecorder.recordInvalidChecksums();
                throw e;
            } catch (InvalidRecordException e) {
                metricsRecorder.recordInvalidChecksums();
                throw new CorruptRecordException(e.getMessage() + " in topic partition " + topicPartition);
            }
```

Reply:
Done. Thanks for this suggestion.